openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java

```diff
@@ -1094,16 +1133,72 @@ public Object invokeMBeanOperation(MBeanServerConnection mbsc, ObjectName object
 			if (rmbe.getCause() instanceof IllegalArgumentException) {
 				Report.printlnInformationMsg(
 						"Invocation failed: IllegalArgumentException (probably guessed arguments badly)");
+				rmbe.printStackTrace();
```

Comment:
Should we fail the test in this case as well?

Reply:
Are you saying you don't think we should fail?  I can't see how we will ever know the condition is being hit unless the test fails.  The test pipelines would 'pass' so the detailed test logs (where the message is written) would never be examined.

The test thinks it has generated suitable arguments for the mxbean operation being invoked - so if that is not the case either something has changed (the argument values are no longer suitable) or there is a new mxbean operation / signature which the test is not (yet) aware of, so failing should be a good way to alert us to any such changes.

Reply:
I was wondering if we should add an Assert.fail(..) over there. Currently it's only printing the stack trace, but not failing the test, in that particular case.

Reply:
Ah - sorry, I assumed your comment applied to the code I had changed (where I added an Assert.fail() (line 1104).

I did not change th e code you mention - but since the RuntimeMBeanException is now caught in a try - catch block specific to the `invoke` method (line 1094) I believe the code in the `catch RuntimeMBeanException` block on line 1131 is now unreachable and should have been removed.  Please check if you agree - if so. I'll update the PR.